src/beacon.js

```diff
@@ -48,14 +48,44 @@ export default function Emitter(endpoint, context, options) {
   let messages = [];
   let timer;
 
-  function send(url, data, sync) {
-    if (typeof window !== 'undefined' && window.navigator.sendBeacon) {
-      // Chrome does not yet support this
-      // const encoded = new Blob([data], { type: 'application/json; charset=UTF-8' });
-      // return window.navigator.sendBeacon(url, encoded);
-      return window.navigator.sendBeacon(url, data);
+  const prepData = function(data) {
+    // iterate through data keys and uri encode any objects
+    const preppedData = {};
+    let parsedData = JSON.parse(data);
+    for (let key in parsedData) {
+      if (typeof parsedData[key] === 'object') {
+        preppedData[key] = JSON.stringify(parsedData[key]);
+      } else {
+        preppedData[key] = parsedData[key];
+      }
+    }
+
+    return preppedData;
+  }
```

Comment:
`JSON.parse()` has a `reviver` argument and `JSON.stringify()` has a `replacer` argument that might be valuable here.